redfish-core/lib/log_services.hpp

```diff
@@ -3346,4 +3415,931 @@ inline void requestRoutesPostCodesEntry(App& app)
+ *        by using the given isolated dbus object.
+ *
+ * @param[in] asyncResp - The redfish response to return.
+ * @param[in] entryJsonIdx - The json entry index to add isolated hardware
+ *                            details. If passing less than or equal 0 then,
+ *                            it will assume the given asyncResp jsonValue as
+ *                            a single entry json object. If passing greater
+ *                            than 0 then, it will assume the given asyncResp
+ *                            jsonValue contains "Members" to fill in the
+ *                            appropriate entry json object.
+ * @param[in] dbusObjIt - The DBus object which contains isolated hardware
+                         details.
+ *
+ * @return The redfish response with appropriate redfish properties of the
+ *         isolated hardware details into LogEntry schema if success else
+ *         failure response.
+ */
+inline void fillSystemHardwareIsolationLogEntry(
+    const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+    const size_t entryJsonIdx, GetManagedObjectsType::const_iterator& dbusObjIt)
+{
+    nlohmann::json& entryJson =
+        (entryJsonIdx > 0
+             ? asyncResp->res.jsonValue["Members"][entryJsonIdx - 1]
+             : asyncResp->res.jsonValue);
+
+    bool entryIsPresent{false};
+    for (auto& interface : dbusObjIt->second)
+    {
+        if (interface.first == "xyz.openbmc_project."
+                               "HardwareIsolation.Entry")
+        {
+            entryIsPresent = true;
+            for (auto& property : interface.second)
+            {
+                if (property.first == "Severity")
+                {
+                    if (auto severity =
+                            std::get_if<std::string>(&property.second))
+                    {
+                        if (*severity == "xyz.openbmc_project."
+                                         "HardwareIsolation.Entry.Type."
+                                         "Critical")
+                        {
+                            entryJson["Severity"] = "Critical";
+                        }
+                        else if ((*severity == "xyz.openbmc_project."
+                                               "HardwareIsolation."
+                                               "Entry.Type.Warning") ||
+                                 (*severity == "xyz.openbmc_project."
+                                               "HardwareIsolation."
+                                               "Entry.Type.Manual"))
+                        {
+                            entryJson["Severity"] = "Warning";
+                        }
+                    }
+                }
+                else if (property.first == "Resolved")
+                {
+                    if (auto resolved = std::get_if<bool>(&property.second))
+                    {
+                        entryJson["Resolved"] = *resolved;
```

Comment:
check nullptr here

Reply:
> check nullptr here

Not required, since the implicit conversion will happen that will return true for non-null pointer and false for a null pointer in the if block itself.

Reply:
Can we follow https://github.com/openbmc/bmcweb/blob/8b6a35f0f99ed38293f7d134511d6181c88c5a59/redfish-core/lib/log_services.hpp#L1397 as close as we can here